addon/components/smart-link-to.js

```diff
@@ -0,0 +1,24 @@
+import { getOwner } from '@ember/application';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+
+export default class SmartLinkToComponent extends Component {
+  @service router;
+
+  @tracked tagName = '';
+  @tracked cssClass = null;
+  @tracked target = null;
+
+  get isRoute() {
+    return getOwner(this).hasRegistration(`route:${this.args.link}`);
+  }
+
+  constructor() {
+    super(...arguments);
```

Comment:
```suggestion
  constructor(owner, args) {
    super(owner, args);
```
https://api.emberjs.com/ember/release/modules/@glimmer%2Fcomponent in constructor section